File: java/IEC62325/MarketOperations/MarketSystem/ExternalInputs/ContingencyConstraintLimit.java
package IEC62325.MarketOperations.MarketSystem.ExternalInputs;

import IEC61970.Base.Core.Curve;
import IEC62325.MarketOperations.ReferenceData.MktContingency;

/**
 * Possibly time-varying max MW or MVA and optionally Min MW limit or MVA limit
 * (Y1 and Y2, respectively) assigned to a constraint for a specific contingency.
 * Use CurveSchedule XAxisUnits to specify MW or MVA.
 * @created 03-Jan-2024 2:08:10 PM
 */
public class ContingencyConstraintLimit extends Curve {

	public MktContingency MktContingency;
	public MWLimitSchedule MWLimitSchedules;
	public SecurityConstraintSum SecurityConstraintSum;

	public ContingencyConstraintLimit(){

	}
}//end ContingencyConstraintLimit

File: py/IEC62325/MarketOperations/MarketSystem/ExternalInputs/ContingencyConstraintLimit.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Wed Dec 27 15:35:30 2023

from IEC61970.Base.Core.Curve import Curve
from IEC62325.MarketOperations.MarketSystem.ExternalInputs.MWLimitSchedule import MWLimitSchedule
from IEC62325.MarketOperations.MarketSystem.ExternalInputs.SecurityConstraintSum import SecurityConstraintSum
from IEC62325.MarketOperations.MarketSystem.ExternalInputs.TransferInterfaceSolution import MktContingency


class ContingencyConstraintLimit(Curve):

    def __init__(self) -> None:
        super().__init__()
        self._mkt_contingency: MktContingency = MktContingency()
        self.mw_limit_schedules: MWLimitSchedule = MWLimitSchedule()
        self.security_constraint_sum: SecurityConstraintSum = SecurityConstraintSum()
